returnn/torch/engine.py

```diff
@@ -130,14 +152,23 @@ def init_train_from_config(
         self._load_model(epoch=self._start_epoch)
         self._save_model_epoch_interval = config.int("save_interval", 1)
 
-        self._updater = Updater(self.config, self._pt_model, self.learning_rate)
+        if self._use_torch_distributed:
+            # wrap the model use torch distributed class
+            self._ddp_pt_model = self._torch_distributed_class(
+                self._pt_model, device_ids=[int(os.environ["LOCAL_RANK"])], **self._torch_distributed_options
```

Comment:
After discussion:

The reason `os.environ` was used is because this was the only known way to get the local rank (as opposed to the global rank).

It's a bit ambiguous what we should use this, whether it is the local rank, global rank, the device id 0, or what else. It also matter how `CUDA_VISIBLE_DEVICES` is set up, single node vs multi node training, etc. Also how it is run later on, via SGE, mpirun, whatever. Using the local rank is probably reasonable for our current use cases though.

The next question is, how to get the local rank. `torch.distributed` does not provide such function?

So, we agreed to introduce some helper functions (in our `returnn.torch.distributed` module):
```python
def get_device_ids():
  # It depends on the specific setup what to return here,
  # how CUDA_VISIBLE_DEVICES is set up, etc.
  # This is currently a reasonable assumption
  # but we might extend the logic later,
  # or make it configurable.
  return [get_local_rank()]

def get_local_rank():
  # torch.distributed does not seem to provide a function for this.
  # Via mpirun (OpenMPI), this env variable would be set.
  # It should fail with an error otherwise.
  return int(os.environ["LOCAL_RANK"])
```

Reply:
as suggested, I added the helper functions in `returnn.torch.distributed` module